src/Interpreters/TemporaryDataOnDisk.h

```diff
@@ -130,6 +134,19 @@ class TemporaryDataOnDisk : private TemporaryDataOnDiskScope
     typename CurrentMetrics::Metric current_metric_scope = CurrentMetrics::TemporaryFilesUnknown;
 };
 
+struct InputReader
```

Comment:
Let's define as a public it inside `TemporaryFileStream`, so it can be used as `TemporaryFileStream::InputReader` (or even just `TemporaryFileStream::Reader` ?)